add elastic zeus handler

janewangfb opened the pull request:
Summary: as titled. We use a special url schem elasticzeus for elastic zeus so that we dont need to change the public interface of init_process_group.

Differential Revision: D13948151

pietern commented:
The zeus store is internal to FB and it shouldn't be hard coded here.

Instead, If we need to make any of these parameters configurable (e.g. for a one-off rendezvous function instead of inferring one from the `init_method`, or bypassing the rendezvous function and specifying its return values manually) then we can do that.

cc @mrshenli

pietern commented:
@janewangfb Can you update the title/summary of this PR to reflect the new purpose?

There should also be a test for this case to check that the specified store gets used.

janewangfb committed:
add elastic zeus handler (#16746)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16746

as titled. We use a special url schem elasticzeus for elastic zeus so that we dont need to change the public interface of init_process_group.

Reviewed By: aazzolini, soumith

Differential Revision: D13948151

fbshipit-source-id: 434d85a8289e7e8e4aa157e2a7f48197bd1ef0c7
